### PR TITLE
Add docs for poo* and goo*

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: src/Network/Minio/Data.hs [hunk 1]
```diff
@@ -184,21 +184,31 @@ type Object = Text
...
+  -- | Can be used to specify caching behavior along the request/reply chain.
   , pooCacheControl       :: Maybe Text
+  -- | Entity header is used to describe the language(s) intended
+  -- for the audience.
   , pooContentLanguage    :: Maybe Text
+  -- | If you don't specify, Standard is the default storage class.
   , pooStorageClass       :: Maybe Text
+  -- | Headers starting with this prefix are user-defined metadata.
   , pooUserMetadata       :: [(Text, Text)]
+  -- | Control concurreny with setting this value to desried number.
```
Comment (krisis) [inline src/Network/Minio/Data.hs]: We could say "Number of worker threads used to upload an object"

File: src/Network/Minio/Data.hs [hunk 2]
```diff
@@ -184,21 +184,31 @@ type Object = Text
...
+  -- referenced by the Content-Type header field.
   , pooContentEncoding    :: Maybe Text
+  -- | Specifies presentational information for the object.
   , pooContentDisposition :: Maybe Text
+  -- | Can be used to specify caching behavior along the request/reply chain.
   , pooCacheControl       :: Maybe Text
+  -- | Entity header is used to describe the language(s) intended
+  -- for the audience.
   , pooContentLanguage    :: Maybe Text
+  -- | If you don't specify, Standard is the default storage class.
```
Comment (krisis) [inline src/Network/Minio/Data.hs]: We could say "Possible values are Standard and ReducedRedundancy. The storage class is Standard by default (i.e when Nothing is passed)."

File: src/Network/Minio/Data.hs [hunk 3]
```diff
@@ -184,21 +184,31 @@ type Object = Text
...
   , pooStorageClass       :: Maybe Text
+  -- | Headers starting with this prefix are user-defined metadata.
```
Comment (krisis) [inline src/Network/Minio/Data.hs]: "Metadata to store with the object uploaded"

File: src/Network/Minio/Data.hs [hunk 4]
```diff
@@ -184,21 +184,31 @@ type Object = Text
...
+-- | A type alias to represent an Entity-Tag returned by S3-compatible APIs.
 type ETag = Text
 
 -- |
 -- Data type represents various options specified for PutObject call.
 -- To specify PutObject options use the poo* accessors.
 data PutObjectOptions = PutObjectOptions {
+  -- | A standard MIME type describing the format of the contents.
     pooContentType        :: Maybe Text
+  -- | Specifies what content encodings have been applied to the object and thus
```
Comment (krisis) [inline src/Network/Minio/Data.hs]: I think we should describe what this field is meant for and need not explain how Content-Encoding header works with Content-Type.

Comment (harshavardhana) [reply src/Network/Minio/Data.hs]: That is what its meant for @krisis

**Comment by harshavardhana** [comment]
PTAL @krisis @donatello

File: src/Network/Minio/Data.hs [hunk 5]
```diff
@@ -184,21 +184,32 @@ type Object = Text
...
+  -- referenced by the Content-Type header field.
   , pooContentEncoding    :: Maybe Text
+  -- | Set presentational information for the object.
   , pooContentDisposition :: Maybe Text
+  -- | Set to specify caching behavior for the object along the
+  -- request/reply chain.
   , pooCacheControl       :: Maybe Text
+  -- | Set to describe the language(s) intended for the audience.
   , pooContentLanguage    :: Maybe Text
+  -- | Set to possible values are Standard and ReducedRedundancy. The
```
Comment (krisis) [inline src/Network/Minio/Data.hs]: This should be "Set Storage class. Possible values are Standard and ReducedRedundancy ..."

**Pull request merged**